assets/js/data/cart/notify-errors.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { ApiErrorResponse, isApiErrorResponse } from '@woocommerce/types';
+import { createNotice, DEFAULT_ERROR_MESSAGE } from '@woocommerce/base-utils';
+import { decodeEntities } from '@wordpress/html-entities';
+import { dispatch } from '@wordpress/data';
+
+/**
+ * This function is used to notify the user of cart errors.
+ */
+export const notifyErrors = ( error: ApiErrorResponse | null = null ) => {
+	if ( error === null || ! isApiErrorResponse( error ) ) {
+		return;
+	}
+
+	let errorMessage = error.message || DEFAULT_ERROR_MESSAGE;
+
+	if ( error.code === 'invalid_json' ) {
```

Comment:
Maybe a comment here explaining why this check is necessary
```suggestion
	// Replace the generic invalid JSON message with something more user friendly.
	if ( error.code === 'invalid_json' ) {
```